src/helpers/util.js

```diff
@@ -39,5 +39,40 @@ module.exports = {
       ret.push(row)
     }
     return ret
+  },
+
+  /*
+    Binary Search:
+    Assumes that @arg array is sorted increasingly
+    return smallest i such that changes[i] < target does not hold,
+    or changes.length if all elements are < target.
+    It returns the index where target could be inserted to maintain the order of the array
+  */
+  findLowerBound: function (target, array) {
+    return findLowerBoundInternal(target, array, 0, array.length)
+  },
+
+  findLowerBoundValue: function (target, array) {
+    var index = this.findLowerBound(target, array)
+    return array[index]
+  }
+}
+
+function findLowerBoundInternal (target, array, lowerbound, length) {
+  if (array.length === 0) {
+    return 0
+  }
+  var middle
+  while (lowerbound + 1 !== length) {
+    middle = Math.floor((length + lowerbound) / 2)
+    if (array[middle] > target) {
+      length = middle
+    } else if (array[middle] < target) {
+      lowerbound = middle
+      length = array.length
+    } else if (array[middle] === target) {
```

Comment:
ok I see, so i will change the doc associated. and yes we might need to change the name cause this is the behavior we want. if the element we are looking for is on the array we have to return his index.